rero_ils/modules/stats/api.py

```diff
@@ -687,6 +698,495 @@ def loans_of_transaction_library_by_item_location(self,
+            query = query.filter('bool', must=[org_filter])
+        else:
+            # filter by organisation libraries
+            org_filter = Q('terms', **{fields['library']: library_pids})
+            query = query.filter('bool', must=[org_filter])
+
+        # add filter to query
+        if filter:
+            cfg_filter = Q('query_string', query=(filter))
+            query = query.filter('bool', must=[cfg_filter])
+
+        # add filter period to query
+        if period:
+            if period == 'month':
+                # previous month
+                previous_month = datetime.now() - relativedelta(months=1)
+                # last day of previous month
+                previous_month = previous_month + relativedelta(day=31)
+                month = '%02d' % previous_month.month
+                _from = f'{previous_month.year}-{month}-01T00:00:00'
+                _to = f'{previous_month.year}-{month}-{previous_month.day}'
+                _to = f'{_to}T23:59:59'
+            else:
+                previous_year = datetime.now() - relativedelta(months=+12)
+                _from = f'{previous_year.year}-01-01T00:00:00'
+                _to = f'{previous_year.year}-12-31T23:59:59'
+
+            query = query\
+                .filter('range', _created={'gte': _from, 'lte': _to})
+
+        # make aggregations according to distributions and execute query
+        return self._query_aggs(query, fields)
+
+    def _validate_distributions(self, distributions):
+        """Validate distributions of statistics report.
+
+        :param distributions: distributions of the configuration.
+        :return True or raise PasswordValidatorException
+        """
+        # check unicity into distributions keys
+        if (len(set(distributions)) != len(distributions)):
+            raise StatReportDistributionsValidatorException(
+                'Distributions should not be the same.')
+        # check time_range_month and time_range_year are not both
+        # selected
+        if 'time_range_month' in distributions and \
+           'time_range_year' in distributions:
+            raise StatReportDistributionsValidatorException(
+                'Distributions should not be both time_range_month and '
+                'time_range_year')
+        # check different type of location searches are not selected
+        if 'item_location' in distributions and \
+           'location' in distributions:
+            raise StatReportDistributionsValidatorException(
+                'Distributions should not be both item_location and '
+                'location')
+        # check different type of library searches are not selected
+        if 'library' in distributions and \
+           'item_owning_library' in distributions:
+            raise StatReportDistributionsValidatorException(
+                'Distributions should not be both library and '
+                'item_owning_library')
+        return True
```

Comment:
Not really need to return true. If function doens't raised any exception, it's validated